core/systemcontractupgrade/system-contract-upgrade.go

```diff
@@ -0,0 +1,103 @@
+package systemcontractupgrade
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type UpgradeConfig struct {
+	ContractAddr common.Address
+	CommitUrl    string
+	Code         string
+}
+
+type Upgrade struct {
+	UpgradeName string
+	Configs     []*UpgradeConfig
+}
+
+var (
+	mainnetChainID = big.NewInt(90)
+	testnetChainID = big.NewInt(96)
+	devnetChainID  = big.NewInt(1417)
+
+	// genesis contracts
+	validatorContract          = common.HexToAddress("0x0000000000000000000000000000000000001000")
+	slashContract              = common.HexToAddress("0x0000000000000000000000000000000000001001")
+	systemRewardContract       = common.HexToAddress("0x0000000000000000000000000000000000001002")
+	lightClientContract        = common.HexToAddress("0x0000000000000000000000000000000000001003")
+	tokenHubContract           = common.HexToAddress("0x0000000000000000000000000000000000001004")
+	relayerIncentivizeContract = common.HexToAddress("0x0000000000000000000000000000000000001005")
+	relayerHubContract         = common.HexToAddress("0x0000000000000000000000000000000000001006")
+	govHubContract             = common.HexToAddress("0x0000000000000000000000000000000000001007")
+	tokenManagerContract       = common.HexToAddress("0x0000000000000000000000000000000000001008")
+	crossChainContract         = common.HexToAddress("0x0000000000000000000000000000000000002000")
+
+	mainnetUpgrapdeConfig = make(map[int64]Upgrade)
+	testnetUpgrapdeConfig = make(map[int64]Upgrade)
+	devnetUpgrapdeConfig  = make(map[int64]Upgrade)
+)
+
+func init() {
+	// setup mainnet upgrade config
+
+	/* mainnet upgrade config */
+
+	// setup testnet upgrade config
+
+	/* testnet upgrade config */
+
+	// setup testnet upgrade config
+	devnetUpgrapdeConfig[20000] = Upgrade{
```

Comment:
can not just simple define `devnetUpgrapdeConfig[20000]` here.  Because `20000` is treat as fork height here, we must define the fork height on chainConfig, like  dao fork get `DAOForkBlock` in config. Otherwise will break the handshake protocol about forkId.

Reply:
We need define a map like  `map[forkname] []*UpgradeConfig` here. Then define different height for the fork for different network on `Params` file